saleor/checkout/complete_checkout.py

```diff
@@ -58,7 +57,11 @@ def _process_voucher_data_for_order(checkout_info: "CheckoutInfo") -> dict:
     :raises NotApplicable: When the voucher is not applicable in the current checkout.
     """
     checkout = checkout_info.checkout
-    voucher = get_voucher_for_checkout_info(checkout_info, with_lock=True)
+    voucher = (
+        Voucher.objects.all().select_for_update().get(code=checkout_info.voucher.code)
```

Comment:
Do we need `.all()` here?

Reply:
Also, not sure why we don't use the previous function. It has much more logic than one simple query.

Reply:
for example, see how lock is working there. It doesn't call select_for_update always. It lock row only for specific case.